tests/test_apply.py

```diff
@@ -87,53 +88,53 @@ def mimic_execution_for_apply(device):
 
 
 @pytest.mark.parametrize("shots,analytic", [(1000, True)])
-class TestStateApply:
-    """Test application of PennyLane operations to state simulators."""
+class TestApplyPureState:
+    """Test application of PennyLane operations on the pure state simulator."""
 
-    def test_basis_state(self, device, tol):
+    def test_basis_state(self, analytic, shots, tol):
         """Test basis state initialization"""
-        dev = device(4)
+        dev = SimulatorDevice(4, analytic=analytic, shots=shots)
```

Comment:
Note: I had to copy these tests for the mixed state simulator, since the expected output had to be a mixed state. To make these tests run properly, I had to call the device directly, rather than using the `device` fixture (which parametrizes over the pure state and mixed state devices)

Reply:
I think all tests should be of this form!

Reply:
(addition of logically separate code should not break existing tests, it should just mean reduced coverage)

Reply:
Adding mixed state simulators does also break the default tests in the `plugin-template` repo. I quote and expand the following idea I commented in the template repo:
___
But I think we should actually have a way to integration test plugins with code inside pennylane that is kept up to date at all times (it's always hard to keep all the plugins and the template up to date). Maybe we could have a functionality to integration test a plug-in with default.qubit as the normal? 
I.e. something useable as 
```python
@pytest.mark.parametrize(*qml.test.parametrization(MyDeviceClass))
def test(args, kwargs):
    qml.test.case(args, kwargs)
```
where test.case returns a sensible error message when it fails. These tests could then make use of the device properties, i.e. if it supports tensor products or inverses or if it is QubitDevice or Device to test all the functionality we add to PennyLane. And the integration tests of the plugins would then always be up-to-date.